### PR TITLE
Remove old Lua code

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
Removes all old lua code from PowerDNS auth server, and provides codesharing between auth and recursor.

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

**Comment by cmouse** [comment]
Currently stuck at _tostring

The LuaContext used keeps the functions available stored in global table. I have not yet been able to decipher how the mapping between object<=>function name<=>function is done, but it seems to rely on pointer to the proper member function. This makes things difficult. I am **assuming** that there is, somewhere, some way to find out what `object:function` is, and that you can actually figure out this on run-time.

It is possible to get a live pointer to the C++ object in question, but I have not yet found a way to execute ::toString only if there is such member function, **or** figure out how to see if there is a _tostring function mapped for the object or not.

It is probably easier to find out the latter than the first, since this information is kept at global function table stored somewhere.

**Comment by cmouse** [comment]
Please merge !5468 first

**Comment by cmouse** [comment]
Also pending on https://github.com/ahupowerdns/luawrapper/pull/35

**Comment by cmouse** [comment]
Not sure why the test fails, I cannot make it fail locally.

**Comment by cmouse** [comment]
The reason the tests fail is that recursor test uses pdns auth server 4.0.4, which is obviously not going to include the lua changes done here. I am not sure how to resolve this problem, please advise.

File: pdns/lua-auth4.cc [hunk 1]
```diff
@@ -136,6 +136,9 @@ bool AuthLua4::axfrfilter(const ComboAddress& remote, const DNSName& zone, const
 
 
 bool AuthLua4::updatePolicy(const DNSName &qname, QType qtype, const DNSName &zonename, DNSPacket *packet) {
+  // default decision is all goes
```
Comment (Habbie) [inline pdns/lua-auth4.cc]: Does this need a documentation update?

Comment (cmouse) [reply pdns/lua-auth4.cc]: no, it's just explicitly documented here.

**Comment by Habbie** [comment]
In the "ext/luawrapper: Synchronize with upstream" commit message, can you clarify synchronize? If it is a full sync, please mention the commit you synced with. If it's an import of one patch, please mention that.

**Pull request merged**